addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -51,28 +46,26 @@ def _compute_l10n_latam_manual_document_number(self):
     def _is_manual_document_number(self, journal):
         return True if journal.type == 'purchase' else False
 
-    @api.depends('name')
+    @api.depends('ref')
     def _compute_l10n_latam_document_number(self):
-        recs_with_name = self.filtered(lambda x: x.name != '/')
-        for rec in recs_with_name:
-            name = rec.name
+        recs_with_ref = self.filtered('ref')
+        for rec in recs_with_ref:
+            ref = rec.ref
             doc_code_prefix = rec.l10n_latam_document_type_id.doc_code_prefix
-            if doc_code_prefix and name:
-                name = name.split(" ", 1)[-1]
-            rec.l10n_latam_document_number = name
-        remaining = self - recs_with_name
+            if doc_code_prefix and ref:
+                ref = ref.split(" ", 1)[-1]
+            rec.l10n_latam_document_number = ref
+        remaining = self - recs_with_ref
         remaining.l10n_latam_document_number = False
```

Comment:
Using `ref` and testing `rec.ref` instead of `rec.name != '/'`, you can simplify the code
```python
for rec in self:
    ref = rec.ref
    if ref and rec.l10n_latam_document_type_id.doc_code_prefix:
        ref = ref.split(" ", 1)[-1]
    rec.l10n_latam_document_number = ref
```